lib/dialects/abstract/query-generator.js

```diff
@@ -2216,13 +2216,18 @@ const QueryGenerator = {
       return;
     }
 
+    item = this.toJSONValue(item);
     items.push(this.whereItemQuery(this._castKey(pathKey, item, cast), {[Op.eq]: item}));
   },
 
-  _castKey(key, value, cast) {
+  toJSONValue(value) {
```

Comment:
make private `_toJSONValue`? I understand you are following `toBooleanValue` but I would like to keep these new methods private